File: app/src/main/java/com/nacare/capture/data/service/network/RetrofitBuilder.java
package com.nacare.capture.data.service.network;


import android.content.Context;

import com.nacare.capture.data.model.FormatterClass;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitBuilder {

    public static Retrofit getRetrofit(Context context, String BASE_URL) {

        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

        FormatterClass formatter = new FormatterClass();
        String username = formatter.getSharedPref("username", context);
        String password = formatter.getSharedPref("password", context);

        BasicAuthInterceptor basicAuthInterceptor = new BasicAuthInterceptor(username, password);

        OkHttpClient client = new OkHttpClient.Builder()
                .readTimeout(2, TimeUnit.MINUTES)
                .connectTimeout(2, TimeUnit.MINUTES)
                .addInterceptor(interceptor)
                .addInterceptor(basicAuthInterceptor)
                .build();

        return new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .client(client)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
    }
}

File: app/src/main/java/com/capture/app/network/RetrofitBuilder.kt
package com.capture.app.network


import android.content.Context
import com.capture.app.data.FormatterClass
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit


object RetrofitBuilder {

    fun getRetrofit(context: Context, BASE_URL: String): Retrofit {

        val formatter = FormatterClass()
        val encodedString = formatter.getSharedPref("encodedString", context)

        val basicAuthInterceptor = BasicAuthInterceptor(encodedString.toString())
        val interceptor = HttpLoggingInterceptor()
        interceptor.level = HttpLoggingInterceptor.Level.BODY

        val client =
            OkHttpClient.Builder()
                .readTimeout(2, TimeUnit.MINUTES)
                .connectTimeout(2, TimeUnit.MINUTES)
                .addInterceptor(interceptor)
                .addInterceptor(basicAuthInterceptor)
                .build()


        return Retrofit.Builder()
            .baseUrl(BASE_URL)
            .client(client)
            .addConverterFactory(GsonConverterFactory.create())
            .build() //Doesn't require the adapter
    }

}